### PR TITLE
Document storage

### LINKED ISSUE #6737: Document storage API with docstring
We should document all the public functions in LocalEnsembleAccessor/LocalExperimentAccessor, etc. with docstrings.

### PR DESCRIPTION
**Pull request by aronhoyer**
**Issue**
Resolves #6737 

**Approach**
_Short description of the approach_

(Screenshot of new behavior in GUI if applicable)

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Make sure tests pass locally (after every commit!)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

**Comment by oyvindeide** [comment]
Did you look into the scikit learn documentation for example? It has some good examples of developer documentation, for example: https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/__init__.py and https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/metrics/_ranking.py

**Review by dafeda — APPROVED** [review]
